hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1093,7 +1209,13 @@ def new_shrinker(self, example, predicate=None, allow_transition=None):
             in_target_phase=self._current_phase == "target",
         )
 
-    def cached_test_function(self, buffer, *, error_on_discard=False, extend=0):
+    def cached_test_function(
+        self,
+        buffer: Union[List[int], bytes, bytearray],
```

Comment:
While the early conversion to `bytes()` means that we _could_ accept any `Iterable[int]` here, and we do actually pass lists from a few test functions, I think the semantics are much clearer if we annotate as only accepting bytes or bytearray.